lib/kafka-consumer.js

```diff
@@ -124,15 +125,24 @@ function KafkaConsumer(conf, topicConf) {
   this.globalConfig = conf;
   this.topicConfig = topicConf;
 
-  this._consumeTimeout = 1000;
+  this._consumeTimeout = DEFAULT_CONSUME_TIME_OUT;
+  this._retryReadInterval = DEFAULT_RETRY_READ_INTERVAL;
 }
 
 /**
  * Set the default consume timeout provided to c++land
  * @param {number} timeoutMs - number of milliseconds to wait for a message to be fetched
  */
 KafkaConsumer.prototype.setDefaultConsumeTimeout = function(timeoutMs) {
-  this._consumeTimeout = timeoutMs;
+  this._consumeTimeout = Number(timeoutMs) || DEFAULT_CONSUME_TIME_OUT;
```

Comment:
Passing `0` here will not work, also you should not wrap interval within `Number(..)` because:
a) validation is done in c++ code
b) it's kind of breaking change because calling `setDefaultConsumeTimeout('anything but number')` will suddenly start working

Maybe something like `this._consumeTimeout = timeoutMs === 0 ? timeoutMs : timeoutMs || DEFAULT_CONSUME_TIME_OUT;` would be safer.

Reply:
Set the value of this._consumeTimeout to timeoutMs directly.